### PR TITLE
Stricter `check_option_description` for single-letter words

### PR DESCRIPTION
**Pull request by omertuc**
```
/// A parameter
```

Would be accepted as a parameter description even though it starts with
an uppercase letter, because argh would consider the word "A" to be an
initialism, which it's not

### CODE REVIEW

File: argh_derive/src/parse_attrs.rs [hunk 1]
```diff
@@ -467,7 +467,7 @@ fn check_option_description(errors: &Errors, desc: &str, span: Span) {
         (Some(x), _) if x.is_lowercase() => {}
         // If both the first and second letter are not lowercase,
         // this is likely an initialism which should be allowed.
-        (Some(x), Some(y)) if !x.is_lowercase() && !y.is_lowercase() => {}
+        (Some(x), Some(y)) if !x.is_lowercase() && (y.is_alphanumeric() && !y.is_lowercase()) => {}
```
Comment (omertuc) [inline argh_derive/src/parse_attrs.rs:470]: `is_alphanumeric` and not `is_alphabetic` to support initialisms like I2C

Comment (mwcz) [reply argh_derive/src/parse_attrs.rs:470]: Seems to me that initialisms like I2C are already accepted:

```
!'2'.is_lowercase() = true
'2'.is_alphanumeric() && !'2'.is_lowercase() = true
```

Comment (omertuc) [reply argh_derive/src/parse_attrs.rs:470]: Yep, they are (even edited my message to say "still"). I'm just justifying my use of `is_alphanumeric` over `is_alphabetic`, which would prevent initialisms like I2C

**Comment by omertuc** [comment]
Worth noting that this is technically a "breaking change" for anyone who happens to already have those in their structs

**Review by sadmac7000 — APPROVED** [review]

**Review by sadmac7000 — CHANGES_REQUESTED** [review]
This probably needs unit tests.

**Review by sadmac7000 — APPROVED** [review]

**Pull request merged**